base/linalg/special.jl

```diff
@@ -6,8 +6,38 @@
 convert{T}(::Type{Bidiagonal}, A::Diagonal{T})=Bidiagonal(A.diag, zeros(T, size(A.diag,1)-1), true)
 convert{T}(::Type{SymTridiagonal}, A::Diagonal{T})=SymTridiagonal(A.diag, zeros(T, size(A.diag,1)-1))
 convert{T}(::Type{Tridiagonal}, A::Diagonal{T})=Tridiagonal(zeros(T, size(A.diag,1)-1), A.diag, zeros(T, size(A.diag,1)-1))
-convert(::Type{LowerTriangular}, A::Bidiagonal) = !A.isupper ? LowerTriangular(full(A)) : throw(ArgumentError("Bidiagonal matrix must have lower off diagonal to be converted to LowerTriangular"))
-convert(::Type{UpperTriangular}, A::Bidiagonal) = A.isupper ? UpperTriangular(full(A)) : throw(ArgumentError("Bidiagonal matrix must have upper off diagonal to be converted to UpperTriangular"))
+
+# methods for conversion from Bidiagonal to [Unit](Upper|Lower)Triangular
+function convert(::Type{LowerTriangular}, A::Bidiagonal)
+    if A.isupper
+        throw(ArgumentError("upper Bidiagonal matrices cannot be converted to LowerTriangular"))
```

Comment:
It could, which segues to a related issue.

On the one hand, existing {constructors for annotation types} that accept an `A<:AbstractMatrix` as the primary argument do not require that `A` 'conform' to the annotation. For example, `LowerTriangular(Bidiagonal(rand(5), rand(4), true))` and `Symmetric(rand(5,5))` are always valid.

On the other hand, `convert(::Type{$(AnnotationType)}, A::$(SpecialMatrixType))`-like methods inconsistently require that `A` 'conform' to the annotation type either in storage structure (e.g. as in the code you identified) or in value (e.g. as in your suggestion). And equivalent `convert` methods from other storage types (e.g. `Matrix`, `SparseMatrixCSC`) to annotation types do not appear to exist. For example, `convert(LowerTriangular, rand(4, 4))` throws a `MethodError`.

Do `convert` methods from special matrix types to annotation types make sense generally? If so, how should they behave? Also if so, shouldn't equivalent `convert` methods from other storage types to annotation types exist? And otherwise, why do these particular methods exist?

Thoughts? Best!